engine/runtime/src/main/java/org/enso/interpreter/node/EnsoRootNode.java

```diff
@@ -49,6 +56,29 @@ protected EnsoRootNode(
     this.sourceLength = sourceSection == null ? NO_SOURCE : sourceSection.getCharLength();
   }
 
+  /** *  Builds a {@link FrameDescriptor} from the alias analysis scope metadata for the local scope.
+    * See [[AliasAnalysis.Graph.Scope.allDefinitions]].
+    *
+    * @return {@link FrameDescriptor} built from the variable definitions in the local localScope.
+    */
+  private static FrameDescriptor buildFrameDescriptor(LocalScope localScope) {
+      var descriptorBuilder = FrameDescriptor.newBuilder();
+      descriptorBuilder.addSlot(FrameSlotKind.Object, LocalScope.monadicStateSlotName(), null);
+      for (var definition : ScalaConversions.asJava(localScope.scope().allDefinitions())) {
+        var returnedFrameIdx =
+          descriptorBuilder.addSlot(
+            FrameSlotKind.Illegal,
+            definition.symbol(),
+            null
+            );
+            // assert(localFrameSlotIdxs(definition.id) == returnedFrameIdx)
+      }
+      descriptorBuilder.defaultValue(DataflowError.UNINITIALIZED);
+      var frameDescriptor = descriptorBuilder.build();
+      // assert(LocalScope.internalSlots.length + localFrameSlotIdxs.size == frameDescriptor.getNumberOfSlots)
```

Comment:
I would suggest to either remove those commented asserts completely or (preferably) bring them in, as they were in the old code

Reply:
Removed in fd3a2286a1